src/plugins/data_source/server/legacy/configure_legacy_client.ts

```diff
@@ -86,20 +117,25 @@ const getQueryClient = async (
  * @returns Legacy client for the given data source endpoint.
  */
 const getRootClient = (
-  dataSourceAttr: DataSourceAttributes,
+  dataSource: DataSourceAttributes,
   config: DataSourcePluginConfigType,
-  { getClientFromPool, addClientToPool }: OpenSearchClientPoolSetup
-): Client => {
-  const endpoint = dataSourceAttr.endpoint;
-  const cachedClient = getClientFromPool(endpoint);
-  if (cachedClient) {
-    return cachedClient as Client;
-  } else {
-    const configOptions = parseClientOptions(config, endpoint);
-    const client = new Client(configOptions);
-    addClientToPool(endpoint, client);
+  { getClientFromPool }: OpenSearchClientPoolSetup
+): Client | undefined => {
+  const {
+    endpoint,
+    auth: { type },
+    lastUpdatedTime,
+  } = dataSource;
 
-    return client;
+  const clientOptions = parseClientOptions(config, endpoint);
+  let cachedClient;
+
+  if (type === AuthType.SigV4) {
+    cachedClient = getClientFromPool(endpoint + lastUpdatedTime, type);
+    return cachedClient ? (cachedClient as Client) : undefined;
```

Comment:
No clientOptions support for awsClient?

Reply:
Good point. Yes, there's no clientOption support for awsClient for now. Because generating awsClient is different flow with other type of clients. We are using the default of Client constructor, but even the existing client option for basic auth/no auth is not fully implemented to support customized config, it's still using defaults. In the future, when we start working on #2009 , we'll make it general to support all auth types